File: hanoi.cpp
#include<iostream>

using namespace std;

int n = 3;
int movimientos = 0;

void hanoi(int n, string inicial, string objetivo, string aux){
    if(n == 0) return;

    hanoi(n - 1, inicial,aux,objetivo); // moviendo los n-1 discos hacia auxiliar

    cout << "disco " << n << " movido desde " << inicial << " hasta " << objetivo << endl; //moviendo el disco n.
    movimientos++;

    hanoi(n - 1, aux,objetivo,inicial); //devolviendo los n - 1 discos desde el auxiliar hasta el objetivo



}

int main(){
    hanoi(n,"inicial","objetivo","aux");
    printf("movimientos %d", movimientos);
}

File: hanoi.py

cont = 0
n = 3

def hanoi(n,inicial,objetivo,aux):
    global cont
    if n == 0:  return

    hanoi(n - 1, inicial,aux,objetivo)
    cont = cont + 1
    print("Disco {} movido desde {} hasta {} ".format(n,inicial,objetivo))
    hanoi(n - 1, aux,objetivo,inicial)


if __name__ == '__main__':
    hanoi(n,"inicial","objetivo","aux")
    print("Movimientos",cont)